### PR TITLE
Product-on-Case template

### PR DESCRIPTION
**Pull request by r4vi**
Show values from good_on_application's firearms details instead of the ones
that come from the `good` directly if available

### CODE REVIEW

File: caseworker/templates/case/product-on-case.html [hunk 1]
```diff
@@ -40,203 +40,205 @@ <h2 class="govuk-heading-l">Full details</h2>
...
+																	{{ good_on_application.good.is_pv_graded|yesno|title }}
+																	{% if good_on_application.good.grading_comment %}
+																			- {{ good_on_application.good.grading_comment }}
+																	{% endif %}
+															</td>
+													</tr>
+													{% if firearm_details %}
+															{% with firearm_types="firearms ammunition components_for_firearms components_for_ammunition" %}
+															{% if firearm_details.type.key in firearm_types.split %}
+																	<tr class="govuk-table__row app-table__row">
```
Comment (saruniitr) [inline caseworker/templates/case/product-on-case.html:110]: these should be for the `firearms` type only

File: caseworker/templates/case/product-on-case.html [hunk 2]
```diff
@@ -40,203 +40,205 @@ <h2 class="govuk-heading-l">Full details</h2>
...
-                                        <td class="govuk-table__cell">
-                                            {% if good_on_application.good.firearm_details.is_replica is not None %}
-                                                {{ good_on_application.good.firearm_details.is_replica|friendly_boolean }}, {{ good_on_application.good.firearm_details.replica_description }}
-                                            {% else %}
-                                                N/A
-                                            {% endif %}
-                                        </td>
-                                    </tr>
-                                {% endif %}
+												{% with firearm_details=good_on_application.firearm_details|default_if_none:good_on_application.good.firearm_details %}
```
Comment (saruniitr) [inline caseworker/templates/case/product-on-case.html:43]: not sure if it is github rendering issue or an indenting issue here

File: caseworker/templates/case/product-on-case.html (hunk 1 above)
Comment (r4vi) [reply caseworker/templates/case/product-on-case.html:110]: I didn't change this - just a whitespace change here

File: caseworker/templates/case/product-on-case.html (hunk 2 above)
Comment (r4vi) [reply caseworker/templates/case/product-on-case.html:43]: I just added the with block before the first occurrence of `firearm_details` and ended it after the last

then did find & replace on both: 

`good_on_application.firearm_details` -> `firearm_details`
`good_on_application.good.firearm_details` -> `firearm_details`

File: caseworker/templates/case/product-on-case.html (hunk 1 above)
Comment (saruniitr) [reply caseworker/templates/case/product-on-case.html:110]: ok, this is the change I have made in other templates, could you please apply below condition for YOM field as well,
`{% if firearm_details.type.key == "firearms" %}`

File: caseworker/templates/case/product-on-case.html (hunk 2 above)
Comment (saruniitr) [reply caseworker/templates/case/product-on-case.html:43]: :+1:

File: caseworker/templates/case/product-on-case.html (hunk 1 above)
Comment (r4vi) [reply caseworker/templates/case/product-on-case.html:110]: thanks I understand - we only want YOM to show up for type=firearms

**Review by saruniitr — APPROVED** [review]

**Pull request merged**